src/main/java/wooteco/subway/service/LineService.java

```diff
@@ -4,49 +4,63 @@
 import java.util.Optional;
 
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 import wooteco.subway.dao.LineDao;
 import wooteco.subway.domain.Line;
-import wooteco.subway.exception.DataNotExistException;
+import wooteco.subway.exception.DataDuplicationException;
+import wooteco.subway.exception.DataNotFoundException;
+import wooteco.subway.service.dto.LineDto;
+import wooteco.subway.service.dto.SectionDto;
 
 @Service
 public class LineService {
 
     private static final int ROW_SIZE_WHEN_NOT_DELETED = 0;
+
     private final LineDao lineDao;
+    private final SectionService sectionService;
+    private final StationService stationService;
 
-    public LineService(LineDao lineDao) {
+    public LineService(LineDao lineDao, SectionService sectionService, StationService stationService) {
         this.lineDao = lineDao;
+        this.sectionService = sectionService;
+        this.stationService = stationService;
     }
 
-    public Line createLine(Line line) {
-        Optional<Line> foundLine = lineDao.findByName(line.getName());
+    @Transactional
+    public Line createLine(LineDto lineDto) {
+        Optional<Line> foundLine = lineDao.findByName(lineDto.getName());
         if (foundLine.isPresent()) {
-            throw new IllegalArgumentException("이미 등록된 노선입니다.");
+            throw new DataDuplicationException("이미 등록된 노선입니다.");
         }
-        return lineDao.save(line);
+        Line newLine = lineDao.save(lineDto.toLine());
+        sectionService.createSection(SectionDto.of(newLine.getId(), lineDto));
+
+        return new Line(newLine,
+            stationService.findBothStationsByIds(lineDto.getUpStationId(), lineDto.getDownStationId()));
```

Comment:
```suggestion
        List<Station> stations = stationService.findBothStationsByIds(lineDto.getUpStationId(), lineDto.getDownStationId())
        return new Line(newLine, stations);
```
한번 끊고 가면 보기에 조금 더 편할 것 같아요! 😄